komoju/komoju-magento
# Add sessions

This PR enables the use of sessions for Magento (using [komoju-php](https://github.com/degica/komoju-php)), replacing the use of the old legacy hosted page. This also means using the API to dynamically grab the payment methods the merchant has enabled on Komoju.

## Testing
1. Build Magento following Tim's instructions and setup the plugin with merchant secret key, etc. (🙇🏻‍♀️ ): [here](https://github.com/degica/komoju-magento/blob/master/docs/dev_setup.md)
2. Test some operations:
- creating an order and completing the komoju session. Ensure the order is then marked as "processing" on Magento admin dashboard
- Make note of the current stock level. Create an order and "cancel" the komoju session (x button). Once it redirects, ensure the original order is then "cancelled" on the Magento admin dashboard and that the stock has returned to the previous level. See that the selected items have been returned to the cart
- Create an order and select Konbini payment method. Cancel it on the komoju dashboard (make sure the webhooks are sent to your Magento endpoint) and verify that the stock for the product has been returned to previous levels and that the order is marked as canceled on the magento admin dashboard
- Go back to the Magento dashboard and update the merchant secret key so that it's wrong. Try placing an order and when you reach the payment page, verify that you see this error:
![image](https://user-images.githubusercontent.com/4694757/105147335-68422080-5b44-11eb-9760-65e1a18fcaf3.png)
(It disappears quite quickly -- that's a native Magento thing AFAIK).
- Any other test cases you can think of

Some notes:
- When a payment method is selected but and someone clicks "place order" but there's some issue connecting to Komoju, the current behaviour cancels the order on Magento, returns the stock, returns the items to the customer's cart, and redirects back to the payment page. I couldn't get the usual way of showing error messages to work from the controller redirect so right now there's no error message.
- ~~Still need to translate the error message from the screenshot above~~
